### PR TITLE
Add more unit test of scaling 

### PR DESCRIPTION
**Pull request by azexcy**
Add more unit test of Scaling 

Changes proposed in this pull request:
-  Add RuleAlteredJobPreparerTest unit test 
-  Increase RuleAlteredJobWorkerTest unit test

### CODE REVIEW

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/source_rules_config.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      input:
+        batchSize: 1000
+        workerThread: 4
+      output:
+        batchSize: 1000
+        workerThread: 4
+      streamChannel:
+        props:
+          block-queue-size: 10000
+        type: MEMORY
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/source_rules_config.yaml:47]: `input/output/streamChannel` is not necessary to configure, we could just keep the required configuration.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/target_rules_config.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      keyGenerateStrategy:
+        column: order_id
+        keyGeneratorName: t_order_snowflake
+      logicTable: t_order
+      shardingStrategy:
+        standard:
+          shardingAlgorithmName: t_order_hash_mod
+          shardingColumn: order_id
+  defaultDatabaseStrategy:
+    standard:
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/target_rules_config.yaml:31]: Could we reuse `target_rules_config.yaml`, seems there's the same content yaml file.

**Review by sandynz — CHANGES_REQUESTED** [review]
Thanks. There is some code style could be improved.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/target_rules_config.yaml (hunk 2 above)
Comment (azexcy) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/resources/scaling/rule_alter/target_rules_config.yaml:31]: The only difference is `- !SHARDING` in the head.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparerTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    
+    @BeforeClass
+    public static void beforeClass() {
+        PipelineContextUtil.mockModeConfigAndContextManager();
+    }
+    
+    @Test(expected = PipelineJobCreationException.class)
+    public void assertPrepareFailedOfNoPrimaryKey() {
+        new RuleAlteredJobPreparer().prepare(new RuleAlteredJobContext(JobConfigurationBuilder.createJobConfiguration()));
+    }
```
[the remark this answers is not in the document]
Comment (azexcy) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparerTest.java]: Add TODO, the method is combined, not suit for simple unit test

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java [hunk 4]
```diff
@@ -18,17 +18,40 @@
...
+import java.util.Optional;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
 public final class RuleAlteredJobWorkerTest {
     
+    static {
+        ShardingSphereServiceLoader.register(ShardingRuleAlteredDetector.class);
+    }
```
[the remark this answers is not in the document]
Comment (azexcy) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java]: no need, remove it.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java [hunk 5]
```diff
@@ -44,7 +67,47 @@ public void assertCreateRuleAlteredContextNoAlteredRule() {
...
+    public void assertHasUncompletedJob() throws IOException, InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        final JobConfiguration jobConfiguration = JobConfigurationBuilder.createJobConfiguration();
+        RuleAlteredJobContext jobContext = new RuleAlteredJobContext(jobConfiguration);
+        JobProgress finishProcess = new JobProgress();
+        finishProcess.setStatus(JobStatus.FINISHED);
+        jobContext.setInitProgress(finishProcess);
+        GovernanceRepositoryAPI repositoryAPI = PipelineAPIFactory.getGovernanceRepositoryAPI();
+        repositoryAPI.persistJobProgress(jobContext);
+        URL jobConfigUrl = getClass().getClassLoader().getResource("scaling/rule_alter/scaling_job_config.yaml");
+        assertNotNull(jobConfigUrl);
```
[the remark this answers is not in the document]
Comment (azexcy) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java:98]: I try to use JobInfo, but the jobParameter is hard coding, so i use the minimum file

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java [hunk 6]
```diff
@@ -44,7 +67,47 @@ public void assertCreateRuleAlteredContextNoAlteredRule() {
     @Test
     public void assertCreateRuleAlteredContextSuccess() {
         JobConfiguration jobConfig = JobConfigurationBuilder.createJobConfiguration();
-        final RuleAlteredContext ruleAlteredContext = RuleAlteredJobWorker.createRuleAlteredContext(jobConfig);
-        Assert.assertNotNull(ruleAlteredContext.getOnRuleAlteredActionConfig());
+        RuleAlteredContext ruleAlteredContext = RuleAlteredJobWorker.createRuleAlteredContext(jobConfig);
+        assertNotNull(ruleAlteredContext.getOnRuleAlteredActionConfig());
```
[the remark this answers is not in the document]
Comment (azexcy) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java]: Embedded

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (sandynz) [reply shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobWorkerTest.java:98]: OK, keep it for now.

**Review by sandynz — APPROVED** [review]

**Pull request merged**